### PR TITLE
test(gtest): show bug with reservations in `gtest`

### PR DESCRIPTION
**Pull request by shamilsan**
Showcase for https://github.com/gear-tech/gear/issues/3699

@gear-tech/dev

### CODE REVIEW

**Review by NikVolf — COMMENTED** [review]
Should we fix it right away?

**Comment by breathx** [comment]
> Should we fix it right away?

As I see it's not 1-line fix, so I guess better to keep an issue open changing its urgency to asap and merge showcase

File: examples/custom/src/reserver.rs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    use gstd::{msg, prelude::*, ReservationId};
+
+    #[derive(Default)]
+    pub(crate) struct State {
+        reservation_id: Option<ReservationId>,
+    }
+
+    pub(crate) fn init() -> State {
+        Default::default()
+    }
```
Comment (breathx) [inline examples/custom/src/reserver.rs:30]: useless

**Review by breathx — APPROVED** [review]

File: examples/custom/src/reserver.rs (hunk 1 above)
Comment (shamilsan) [reply examples/custom/src/reserver.rs:30]: It is not the `init` entrypoint. This function returns state to be used here: 

https://github.com/gear-tech/gear/pull/3698/files#diff-7338aaa9606870091a816d7873106f089b9cc8465c00bd8dfc476ced27002d55R87

Comment (shamilsan) [reply examples/custom/src/reserver.rs:30]: But we can replace it by creating `State` in place indeed.

Comment (shamilsan) [reply examples/custom/src/reserver.rs]: [af7e257](https://github.com/gear-tech/gear/pull/3698/commits/af7e25764c7fe81df19b160a3d1665be54f0ac3b)

**Review by techraed — APPROVED** [review]

**Pull request merged**